docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_batch_definition/create_a_batch_definition.py

```diff
@@ -0,0 +1,98 @@
+from tests.integration.db.taxi_data_utils import load_data_into_test_database
+
+# add test_data to database for testing
+load_data_into_test_database(
+    table_name="postgres_taxi_data",
+    csv_path="./data/yellow_tripdata_sample_2020-01.csv",
+    connection_string="postgresql+psycopg2://postgres:@localhost/test_ci",
+)
+
+# Set up the Data Source and a Data Asset
+context = gx.get_context()
+
+datasource_name = "my_datasource"
+my_connection_string = "${POSTGRESQL_CONNECTION_STRING}"
+
+data_source = context.data_sources.add_postgres(
+    name=datasource_name, connection_string=my_connection_string
+)
+
+asset_name = "MY_TABLE_ASSET"
+database_table_name = "postgres_taxi_data"
+table_data_asset = data_source.add_table_asset(
+    table_name=database_table_name, name=asset_name
+)
+
+# The example starts here
+# <snippet name="docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_batch_definition/_create_a_batch_definition.md full example">
+import great_expectations as gx
+
+context = gx.get_context()
+# <snippet name="docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_data_asset/create_a_data_asset.py retrieve a Data Asset">
+
+# Retrieve a Data Source
+datasource_name = "my_datasource"
+data_source = context.get_datasource(datasource_name)
+
+# Get the Data Asset from the Data Source
+asset_name = "MY_TABLE_ASSET"
+data_asset = data_source.get_asset(asset_name)
+# </snippet>
+
+# Example of a full table Batch Definition
+# highlight-start
+# <snippet name="docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_batch_definition/_create_a_batch_definition.md full table batch definition">
+full_table_batch_definition = data_asset.add_batch_definition_whole_table(
+    name="FULL_TABLE"
+)
+# </snippet>
+# highlight-end
+
+# Verify that the Batch Definition is valid
+# <snippet name="docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_batch_definition/_create_a_batch_definition.md verify full table">
+full_table_batch = full_table_batch_definition.get_batch()
+full_table_batch.head()
+# </snippet>
+
+# Examples of partitioned Batch Definitions
+# highlight-start
+# <snippet name="docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_batch_definition/_create_a_batch_definition.md daily batch definition">
+date_column = "pickup_datetime"
+daily_batch_definition = data_asset.add_batch_definition_daily(
+    name="DAILY", column=date_column
+)
+monthly_batch_definition = data_asset.add_batch_definition_monthly(
+    name="MONTHLY", column=date_column
+)
+yearly_batch_definition = data_asset.add_batch_definition_yearly(
+    name="YEARLY", column=date_column
+)
+# </snippet>
+# highlight-end
+
+# Verify that the partitioned Batch Definitions are valid
+# <snippet name="docs/docusaurus/docs/core/connect_to_data/sql_data/_create_a_batch_definition/_create_a_batch_definition.md verify daily">
+daily_batch = daily_batch_definition.get_batch(
+    batch_parameters={"year": "2020", "month": "01", "day": "14"}
```

Comment:
Is it correct to use strings for the numbers here? I the demos I've done of sql, I've used integers. IIRC, strings are needed for file-based because of substitutions in filenames, but that's kind of a bummer to me that users have to do that. Either way, probably best to use numbers in this case.

Reply:
Change made.  We'll have to do a second pass when we're putting these scripts under test (I'm temporarily removing them from the Integration Test Fixtures since I'm still getting that error around incorrectly formatted data, and we'll be making a second pass to ensure all the examples are in scripts and under test.)